### ISSUE TITLE
New Rule: Tailscale API Key (@secretlint/secretlint-rule-tailscale)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1533]
## Description

Tailscale API keys have a documented `tskey-` prefix with a structured format that makes them uniquely identifiable with zero false positives. GitHub Secret Scanning supports these as Push Protection targets.

Leaked Tailscale keys can grant access to manage devices, ACLs, and network configuration.

## Confidence

The NG samples in this issue were detected by **GitHub Secret Scanning as real "Public leak" alerts** immediately after this issue was published (covering `tskey-api-`, `tskey-auth-`, and `tskey-oauth-` variants). This confirms:

- The proposed regex matches the same pattern GitHub uses for Push Protection
- Real Tailscale keys would be reliably detected
- High-confidence rule candidate with zero expected false positives

## Official Source

- Tailscale official documentation: https://tailscale.com/kb/1101/api
- GitHub Push Protection: supported (listed under "Tailscale" provider)
- TruffleHog: has detection rule (gitleaks does not currently have one)

The `tskey-` prefix is officially documented by Tailscale. Keys follow the format `tskey-{type}-{identifier}-{secret}`.

### API Key / Auth Key / OAuth Key

- **Message ID:** `TAILSCALE_KEY`
- **Prefix:** `tskey-`
- **Structure:** `tskey-{type}-{part1}-{part2}` where type is `api`, `auth`, or `oauth`
- **Character set:** `[a-zA-Z0-9_-]`
- **Regex:**

```regex
/(?<!\p{L})tskey-[a-z]{2,10}-[A-Za-z0-9_]{1,50}-[A-Za-z0-9_]{1,50}(?!\p{L})/gu
```

## OK Examples (should NOT be detected)

```
tskey-
tskey-x
my-tskey-not-real
```

## NG Examples (should be detected)

> Note: the example values below are dummies (verified by GitHub Secret Scanning as matching the production pattern).

```
tskey-api-AAAAAAAAAAAA-BBBBBBBBBBBBBBBBBBBBBBBB
tskey-auth-CCCCCCCCCCCC-DDDDDDDDDDDDDDDDDDDDDDDD
tskey-oauth-EEEEEEEEEEEE-FFFFFFFFFFFFFFFFFFFFFFFF
```
